docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+         * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+    * In order to keep things running smoothly, if a breaking change requires serious review from the consumer teams, the PRs for that change should be approved and ready to merge before that breaking change is merged into application-services.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+    * Manually merge our code whenever we make changes to a `beta`, `release`, or `esr-*` branch.
+       * Commit a version bump to `[firefox-version].[N+1]`
+       * Make a PR in the corresponding branch for each application.  Create a tool to automate this.
+       * Collaborate with the mobile teams to get the PR merged.
+       * Merge the changes forward into the other branches.  If the change was committed to `release`, then we merge it into `beta`, and `main`.
+    * Continue to maintain our CHANGELOG, but with new versioning.  In main, we keep a rolling log of updates with the header `[firefox-version].0`
+    * On the monthly Firefox merge date, we do roll each branch forward:
+      - Create a new `esr-[version]` branch from `release`, if needed
+      - Merge `beta` into `release` (this should be a fast-forward).
+      - Bump the version number in main to `[firefox-version].0`
+      - Merge `main` into `beta` (also a fast-forward).
+      - Start a new changelog header in `main`
+    * Note: these things replace our current release process -- no more cutting releases.
+    * Coordinate this all with release engineering and see if they want to be involved in any of these steps.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it gives us more time to observe other repos moving into `firefox-android` before we take action.
+
+### Move to a rapid release process where changes in app-services are frequently merged into consumer applications
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Bad, because we need implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
+  * Good, because this will unblock future changes from being merged and released on that application.
```

Comment:
I don't quite understand this point, maybe we can elaborate a bit more?

Reply:
The previous 2 points work together, maybe I should merge them into a single bullet somehow.

My main point is that we're going to be losing some flexibility, for example we couldn't do what we did with logins where ios took a long time to take the changes.  However, that flexibility has a huge downside: it makes it really difficult to merge in any new code in the meantime.  So it's both good and bad, but I believe it's good overall.  That's what I was trying to get at with the "We are willing to accept the trade-offs..." decision driver.